### PR TITLE
Put benchmarks into the required build target

### PR DESCRIPTION
**Pull request by angerman**
this should make the build fail if any benchmark fails.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

File: release.nix [hunk 1]
```diff
@@ -218,8 +218,24 @@ in pkgs.lib.fix (jobsets: mapped // mapped-nix-tools' // {
...
+      # exes are nix-tools.exes.$pkg -- which will contain *all* executables of the relevant pkg
       (builtins.concatLists (lib.attrValues (lib.mapAttrs (_: allLinux) jobsets.nix-tools.exes)))
+      # benchmarks are nix-tools.benchmarks.$pkg.$benchmark
+      (builtins.concatLists
+       (builtins.concatLists
+        (lib.attrValues
+         (lib.mapAttrs
+          (name: value: lib.attrValues (lib.mapAttrs (_: allLinux) value))
+          jobsets.nix-tools.benchmarks))))
+#      # tests are nix-tools.tests.$pkg.$test
```
Comment (angerman) [inline release.nix]: I don't really understand why this doesn't work. But it doesn't :( It should be very similar to the benchmarks section.

**Comment by rvl** [comment]
Rather than using the nix-tools component derivations for building benchmarks, you could use the old stack2nix derivations.
Basically, it got broken by f1e2418717b5618b078442a52e4ef230a04d1919 (PR #3877). With a fix for #4023, I _think_ it should be possible to default `enableBenchmarks = true` again.

**Comment by angerman** [comment]
@rvl, I guess the question is if we want to keep the old passes. But for cardano-sl it might not matter that much anymore.

**Comment by angerman** [comment]
Best part is that this PR is now blocked on broken benchmarks 😭

**Pull request closed without merging**